search-app/src/cmr/search/services/humanizers/humanizer_report_service.clj

```diff
@@ -70,18 +75,18 @@
   (map #(rfm->umm-collection context %) rfms))
 
 (defn- wait-and-retry-for-collection-cache
-  "Wait configurable number of seconds before retrying to get all the collections
-   from collection cache. If collections are still not present, throw warning."
-  [context]
-  (info (format "Collection cache is not populated after %d seconds of delay" 
-                (humanizer-report-generator-job-delay)))
-  (Thread/sleep (humanizer-report-generator-job-wait)) 
-  (let [rfms (metadata-cache/all-cached-revision-format-maps context)]
-    (if (seq rfms)
-      rfms
-      (warn (format "Collection cache is not populated after %d seconds of delay and %d milli-seconds of wait."
-                    (humanizer-report-generator-job-delay)
-                    (humanizer-report-generator-job-wait))))))
+  "Wait configurable number of milli seconds before retrying configurable number of times
+   to get all the collections from collection cache."
+  [count context]
+  (when (< count (retry-count))
+    (info (format (str "Humanizer report generator job is sleeping for %d milli-seconds"
+                       "before retrying to see if the collection cache is populated.")
+                  (humanizer-report-generator-job-wait)))
+    (Thread/sleep (humanizer-report-generator-job-wait))
+    (let [rfms (metadata-cache/all-cached-revision-format-maps context)]
+      (if (seq rfms)
+        rfms
+        (recur (inc count) context)))))
 
 (defn- get-all-collections
   "Retrieves all collections from the Metadata cache, partitions them into batches of size
```

Comment:
1. I thought the normal flow would be we call to get the collections from the cache first, if it doesn't exist, we fire the retry mechanism. If you just want to start the retry mechanism right away, I guess it's okay...  but then, we can't say that collection cache is not populated after the 400 seconds delay because we never try it.  I will need to remove that msg. 
2. Your loop is basically the same as mine except you are decreasing the count and I'm increasing the count.